tests/integration-service/integration.go

```diff
@@ -263,6 +265,69 @@ var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests
 			Expect(f.AsKubeAdmin.CommonController.HaveTestsSucceeded(snapshot)).To(BeFalse(), "expected tests to fail for snapshot %s/%s", snapshot.GetNamespace(), snapshot.GetName())
 		})
 
+		It("creates a new IntegrationTestScenario", func() {
+			newIntegrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario(applicationName, testNamespace, BundleURL, InPipelineName)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("updates the Snapshot with the re-run label for the new scenario", FlakeAttempts(3), func() {
+			updatedSnapshot := snapshot.DeepCopy()
+			err := metadata.AddLabels(updatedSnapshot, map[string]string{snapshotRerunLabel: newIntegrationTestScenario.Name})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(f.AsKubeAdmin.IntegrationController.PatchSnapshot(snapshot, updatedSnapshot)).Should(Succeed())
+			Expect(metadata.GetLabelsWithPrefix(updatedSnapshot, snapshotRerunLabel)).NotTo(BeEmpty())
+		})
+
```

Comment:
would it make sense to use function [AddIntegrationTestRerunLabel](https://github.com/redhat-appstudio/integration-service/blob/76a2a6657bcc227a0864038f86105ae4358aea42/gitops/snapshot.go#L718) 
?

Reply:
I thought about it but decided against it since the re-run label will be added in different ways also (e.g. through the UI or even manually) and the current way of testing covers that more reliably.